### ISSUE TITLE
Unexpected end of JSON Input - When using Batching

### ISSUE THREAD

**Issue by simkessy** [SharePoint/PnP-JS-Core#545]
#### Category
[ ] Enhancement

[x ] Bug

[x ] Question

#### Version

Please specify what version of the library you are using: [     latest    ]

When I try to delete a list using batch , I get a console error saying "Unexpected end of JSON Input"

The list get's deleted property but I'm not sure how to eliminate this error. This also occurs during batch updates:

```
deleteLists = function () {
    var b = $pnp.sp.createBatch();
    $pnp.sp.web.lists.getByTitle("Meeting Minutes").inBatch(b).delete();
    $pnp.sp.web.lists.getByTitle("Decisions").inBatch(b).delete();
    $pnp.sp.web.lists.getByTitle("Decision Tasks").inBatch(b).delete();
    $pnp.sp.web.lists.getByTitle("Template Tasks").inBatch(b).delete();
    $pnp.sp.web.lists.getByTitle("Project Task List").inBatch(b).delete();

    return b.execute();
};

project.update = {
    field: null,
    batch: null,
    list: null,
    set: function (listName) {
        project.update.batch = $pnp.sp.createBatch();
        project.update.list = $pnp.sp.web.lists.getByTitle(listName);
        project.update.field = project.update.list.fields;
    },
    fieldName: function (internal, display) {
        console.log("update:", internal, display);
        project.update.field.getByInternalNameOrTitle(internal).inBatch(project.update.batch).update({
            Title: display
        });
    },
    meetingMinutes: function () {
        project.update.set('Meeting Minutes');

        // Renames 
        project.update.fieldName("Title", "Topic");
        project.update.fieldName("Notes", "Agenda/Notes");

        return project.update.batch.execute();
    },
```

![image](https://user-images.githubusercontent.com/5871660/28728634-64539532-7397-11e7-8b00-f0d59485b01a.png)

**Comment by patrick-rodgers** [comment]
What browser and SP environment are you in? I wonder if this is related to #487

**Comment by simkessy** [comment]
@patrick-rodgers  I'm using Chrome (latest) on SharePoint Online.

**Comment by koltyakov** [comment]
Looks like I have the same issue for item delete operations in a batch in SP2016 too.

**Comment by garrytrinder** [comment]
In regards to #487 a quick workaround is to use Recycle() instead of Delete()

**Comment by simkessy** [comment]
I didn't even know that was an option.

**Comment by patrick-rodgers** [comment]
Closing as fixed in #559
